### PR TITLE
fix(report): Cleaning inventory list registry 

### PR DESCRIPTION
**Pull request by DedrickEnc**
This PR fix the inventory registry so that the pdf report will be affected by the filtering made by the user through the search module.

### CODE REVIEW

File: server/controllers/inventory/reports/prices.js [hunk 1]
```diff
@@ -42,29 +38,15 @@ function prices(req, res, next) {
...
-
-  filters.fullText('text', 'text', 'inventory');
-  filters.equals('group_uuid');
-  filters.setOrder('ORDER BY inventory.text');
-
-  const query = filters.applyQuery(sql);
-  const parameters = filters.parameters();
 
-  db.exec(query, parameters)
+  inventorycore.getItemsMetadata(params)
```
Comment (jniles) [inline server/controllers/inventory/reports/prices.js:42]: :+1:

File: client/src/modules/inventory/list/list.js [hunk 2]
```diff
@@ -17,6 +17,8 @@ function InventoryListController(
   GridExport, Languages, Session
 ) {
   var vm = this;
+  vm.download = Inventory.download;
```
Comment (jniles) [inline client/src/modules/inventory/list/list.js:20]: :+1:

File: server/controllers/inventory/reports/prices.js [hunk 3]
```diff
@@ -6,30 +6,26 @@
...
 const ReportManager = require('../../../lib/ReportManager');
+const inventorycore = require('../inventory/core');
 
 module.exports = prices;
 
 const TEMPLATE = './server/controllers/inventory/reports/prices.handlebars';
 
 function prices(req, res, next) {
-  const params = req.query.params ? JSON.parse(req.query.params) : {};
+  const params = _.clone(req.query);
```
Comment (jniles) [inline server/controllers/inventory/reports/prices.js:23]: Nice 👍

File: server/controllers/inventory/reports/prices.handlebars [hunk 4]
```diff
@@ -30,22 +33,25 @@
                 {{ name }}
               </th>
 
-              <th colspan="3" style="border:none; border-bottom: solid black 2px;" class="text-right">
+              <th colspan="6" style="border:none; border-bottom: solid black 2px;" class="text-right">
```
Comment (jniles) [inline server/controllers/inventory/reports/prices.handlebars]: Could you make the `{{ name }}` `<th>` colspan="3" and this one colspan="3"?  That way the name won't be squished so much by the extra columns added here.  See below:
medicinesandcomprimes
_colspan=3 would bring the name's `<th>` out to the red line_

File: server/controllers/inventory/reports/prices.js [hunk 5]
```diff
@@ -6,30 +6,26 @@
...
  * @requires lodash
  * @requires ReportManager
+ * @requires inventorycore
  */
 
 const _ = require('lodash');
-const db = require('../../../lib/db');
-const FilterParser = require('../../../lib/filter');
 const ReportManager = require('../../../lib/ReportManager');
+const inventorycore = require('../inventory/core');
```
Comment (jniles) [inline server/controllers/inventory/reports/prices.js:16]: :+1:  Good resuse of code.

File: client/src/modules/inventory/list/list.html [hunk 6]
```diff
@@ -35,8 +35,7 @@
 
             <li role="separator" class="divider"></li>
             <li role="menuitem">
-              <a ng-href="/reports/inventory/items/?params={{ InventoryCtrl.filters.formatHTTP(true) }}&lang={{ InventoryCtrl.lang }}"
-                download="{{ 'INVENTORY.PRICE_LIST_REPORT' | translate }}">
+              <a ng-href="/reports/inventory/items/?{{ InventoryCtrl.download('pdf') }}" download="{{ 'INVENTORY.PRICE_LIST_REPORT' | translate }}">
```
Comment (jniles) [inline client/src/modules/inventory/list/list.html:38]: Nice 👍

**Review by jniles — COMMENTED** [review]
Looks pretty good!  The reuse of code is excellent and the `download` method is a nice abstraction.

One small thing - could you give the `<th>` that has the inventory name a bit more room?  It's just a change of colspans.  After that, it LGTM.

**Comment by DedrickEnc** [comment]
@jniles
Done!

File: server/controllers/inventory/reports/prices.handlebars [hunk 7]
```diff
@@ -26,26 +29,29 @@
 
             <!-- this is the inventory group header -->
             <tr style="border:none">
-              <th style="border:none; border-bottom: solid black 2px;" class="text-uppercase">
+              <th colspan="3" style="border:none; border-bottom: solid black 2px;" class="text-uppercase">
```
Comment (jniles) [inline server/controllers/inventory/reports/prices.handlebars:32]: :+1:

**Review by jniles — APPROVED** [review]
LGTM!  Thanks!

bors r+

**Pull request merged**